sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosSchedulers.java

```diff
@@ -16,4 +17,11 @@
         COSMOS_PARALLEL_THREAD_NAME,
         Schedulers.DEFAULT_POOL_SIZE,
         true);
+
+    // Custom bounded elastic scheduler to switch off IO thread to process response.
+    public final static Scheduler TRANSPORT_RESPONSE_BOUNDED_ELASTIC = Schedulers.newBoundedElastic(
+        Schedulers.DEFAULT_BOUNDED_ELASTIC_SIZE,
+        Schedulers.DEFAULT_BOUNDED_ELASTIC_QUEUESIZE,
+        TRANSPORT_RESPONSE_BOUNDED_ELASTIC_THREAD_NAME
+    );
```

Comment:
any reason we are using a custom bounded elastic scheduler and not re-using the existing one?

also does this need to be released as a resource when the client is closed?

Reply:
The current one is using parallel schedulers, which is backed by fixed number of workers(currently by default is same as processor count) and has been used in many different places like scheduling retry.

The main reason to create a new one is to not block any retry scheduling etc if the response processing take long time(like the de/serializing for large payload).

When creating a new one, I do not have a strong preferences between parallel vs bounded elastic. Choosing bounded -elastic because parallel is better for short-lived tasks, but for response processing it might not be the case.

Reply:
BoundedElastic would only reuse the workers once the workers have been shut down. Is that what we want?
Are we shutting down the workers after every response processing? If not, I think parallel might be better here. 

BoundedElastic is actually suitable for long running blocking tasks - I am not sure if response processing is one of them. Also we will need to shut down every worker for it to be able to get re-used. Shutting down worker might have extra time hit.

Reply:
Based on the benchmark tests, boundedElastic shows slightly better perf compared to parallel. So will keep using boundedElastic for now.